### PR TITLE
[action] Fix `get_version_number` curly braces

### PR DESCRIPTION
**Pull request by florentmorin**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Fix issue #15194 when `get_version_number` should read build setting variable.

### Description

On issue #15194, when `get_version_number` read build setting variable with curly braces, it didn't work.
Now, it does.

### Testing Steps

Test code was included in `get_version_number_action_spec.rb` and is called `gets the correct version number for 'TargetVariableCurlyBraces'`.
So, It seems to be OK.

### CODE REVIEW

File: fastlane/lib/fastlane/actions/get_version_number.rb [hunk 1]
```diff
@@ -21,6 +21,8 @@ def self.run(params)
         # Get from build settings if needed (ex: $(MARKETING_VERSION) is default in Xcode 11)
         if version_number =~ /\$\(([\w\-]+)\)/
           version_number = get_version_number_from_build_settings!(target, $1, configuration)
+        elsif version_number =~ /\$\{([\w\-]+)\}/
```
Comment (janpio) [inline fastlane/lib/fastlane/actions/get_version_number.rb]: Maybe we can combine these two regexes?
Something like `/\$(\(|\{)([\w\-]+)(\)|\})/` could work maybe.

Comment (florentmorin) [reply fastlane/lib/fastlane/actions/get_version_number.rb]: Yes. But no. This regex [allows](https://rubular.com/r/0xhimXyOFsgu4j) `${MARKETING_VERSION)`

The good regex is `/\$\(([\w\-]+)\)|\{([\w\-]+)\}/` but :

- it's harder to understand _(and to maintain)_
- it's requires double check to extract value from matching groups (`$1` and `$2`), so more code is finally required.

This is why I prefer an easier code than an optimized code.

Comment (janpio) [reply fastlane/lib/fastlane/actions/get_version_number.rb]: Ha, good point - you are totally right.
Maybe add a comment then for the new line a la "${MARKETING_VERSION} also works" or something like that.

Comment (florentmorin) [reply fastlane/lib/fastlane/actions/get_version_number.rb]: OK. Comment just append.

**Review by joshdholtz — APPROVED** [review]
Looks good! Thanks for the contribution 😊❤️

**Pull request merged**